### PR TITLE
State Timeline: Enable support for annotations

### PR DESCRIPTION
**Pull request by dprokop**
This PR brings support for annotations in the State timeline panel, both display and edit possibilities.

### CODE REVIEW

**Comment by codeincarnate** [comment]
Alright, actually have had a little time to test this. The annotation interface is working fine, however annotations don't end up showing up. Looks like either the back-end may be missing them or not picking it up. Looking further.

**Comment by codeincarnate** [comment]
Seeing a few issues at the moment:

- Creating an annotation will cause other annotations to load correctly even with a reload. However navigating away and back will cause annotations to load correctly.
- Time ranges aren't displaying correctly after reload (displays as single point)
- Adding a new annotation causes existing annotations to disappear

I think most of these are related to the back-end not getting every data point necessary for display. I'm looking into it some more.

**Comment by codeincarnate** [comment]
Re-tested this given there have been a fair number of changes in state timeline lately and thankfully that appears to have alleviated the above errors 🥳 

https://user-images.githubusercontent.com/199847/174914207-a26fe212-4082-480d-a978-afdc04cd6944.mov

I believe we may be good to go here but would certainly appreciate a second set of eyes 👀

**Comment by dprokop** [comment]
@codeincarnate thanks for having a look. I'll try to make this workable this week.

**Comment by dprokop** [comment]
Yeah, ready to merge if we get an approval ;D

File: public/app/plugins/panel/state-timeline/StateTimelinePanel.tsx
Comment (mdvictor) [inline public/app/plugins/panel/state-timeline/StateTimelinePanel.tsx:131]: Question: Would we not need to check if annotations creation is enabled like [here](https://github.com/grafana/grafana/blob/main/public/app/plugins/panel/timeseries/TimeSeriesPanel.tsx#L53)?

**Review by mdvictor — APPROVED** [review]
One question regarding annotations creation enabling, otherwise LGTM!

File: public/app/plugins/panel/state-timeline/StateTimelinePanel.tsx
Comment (dprokop) [reply public/app/plugins/panel/state-timeline/StateTimelinePanel.tsx:131]: @mdvictor very good catch! Will add that

**Pull request merged**
